rootfs/scheduler/k8s.py

```diff
@@ -503,35 +535,51 @@ def _get_service(self, name, namespace):
 
         return response
 
+    def _create_secret(self, namespace):
+        secretId, secretKey = '', ''
+        with open("/var/run/secrets/deis/minio/user/access-key-id") as the_file:
+            secretId = the_file.read()
+        with open("/var/run/secrets/deis/minio/user/access-secret-key") as the_file:
+            secretKey = the_file.read()
+        Key, Id = base64.b64encode(secretKey), base64.b64encode(secretId)
+        l = {
+            "version": self.apiversion,
+            "id": namespace,
+            "secretId": Id,
+            "secretKey": Key,
+            }
+        template = json.loads(string.Template(SECRET_TEMPLATE).substitute(l))
+        url = self._api("/namespaces/{}/secrets", namespace)
+        resp = self.session.post(url, json=template)
+        if unhealthy(resp.status_code):
+            error(resp, 'failed to create secret in Namespace "{}"', namespace)
+
     def _create_service(self, name, app_name, app_type, data={}, **kwargs):
         docker_cli = Client(version="auto")
+        image = kwargs.get('image')
         try:
-            image = kwargs.get('image')
+            image = self.registry + '/' + image
             # image already includes the tag, so we split it out here
             docker_cli.pull(image.rsplit(':')[0], image.rsplit(':')[1])
             image_info = docker_cli.inspect_image(image)
             port = int(image_info['Config']['ExposedPorts'].keys()[0].split("/")[0])
         except:
             port = 5000
-
         l = {
             "version": self.apiversion,
             "port": port,
             "type": app_type,
             "name": app_name,
         }
-
         # Merge external data on to the prefined template
         template = json.loads(string.Template(SERVICE_TEMPLATE).substitute(l))
         data = dict_merge(template, data)
-
         url = self._api("/namespaces/{}/services", app_name)
         resp = self.session.post(url, json=data)
         if resp.status_code == 409:
             srv = self._get_service(app_name, app_name).json()
             if srv['spec']['selector']['type'] == 'web':
                 return
-
```

Comment:
All these spaces were intentionally added in because code needs to be readable; One continuous block is not readable.

I'll revamp this in a later PR